components/scream/src/dynamics/homme/atmosphere_dynamics.cpp

```diff
@@ -127,29 +148,27 @@ set_required_group (const FieldGroup<const Real>& group)
 void HommeDynamics::
 set_updated_group (const FieldGroup<Real>& group)
 {
-  if (group.m_info->size()>0) {
-    const auto& name = group.m_info->m_group_name;
-
-    EKAT_REQUIRE_MSG(name=="TRACERS",
-      "Error! We were not expecting a field group called '" << name << "\n");
-
-    EKAT_REQUIRE_MSG(group.m_info->m_bundled,
-        "Error! Shoc expects bundled fields for tracers.\n");
-
-    m_dyn_fields_out["Q"] = *group.m_bundle;
-  } else {
-    // This must be a Homme-standalone run. Allocate a field manually
-    // TODO: add check to make sure this is indeed a standalone run
-    constexpr int qsize_d = HOMMEXX_QSIZE_D;
-    const auto VAR = ShortFieldTagsNames::VAR;
-    auto layout = m_dyn_grid->get_3d_vector_layout(true,VAR,qsize_d);
-    auto nondim = ekat::units::Units::nondimensional();
-    FieldIdentifier Q_fid("Q",layout,nondim,m_dyn_grid->name());
-    Field<Real> Q(Q_fid);
-    Q.get_header().get_alloc_properties().request_allocation<Homme::Scalar>();
-    Q.allocate_view();
-    m_dyn_fields_out.emplace("Q",Q);
-  }
+  const auto& name = group.m_info->m_group_name;
+  EKAT_REQUIRE_MSG(name=="TRACERS",
+    "Error! We were not expecting a field group called '" << name << "\n");
+
+  EKAT_REQUIRE_MSG(group.m_info->size()>0,
```

Comment:
Fun fact: this could also be written

```
EKAT_REQUIRE_MSG(not group.m_info->empty(),
...
```

Reply:
Ah, right. I'll change to that, cause is faster to parse visually.

Reply:
I take it back. `*m_info` is a `FieldGroupInfo` struct, not a std set. I added the `size()` method, to avoid quering the list of names stored internally, but not an 'empty' one. But maybe it's a verbose method worth adding.